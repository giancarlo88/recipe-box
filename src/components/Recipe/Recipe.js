import React from 'react'
import styled from 'styled-components'

const Recipe = styled.div`
  background-color: #247BA0;
`

export default () => <Recipe>Here a recipe will go.</Recipe>
