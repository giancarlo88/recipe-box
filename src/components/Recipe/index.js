import React from 'react'
import styled from 'styled-components'

import ObliqueText from '../ObliqueText'
import RecipeTitle from '../RecipeTitle'

const StyledRecipe = styled.div`
  background-color: #247ba0;
  color: #C9DAEA;
  height: ${(props) => (props.isSelected ? '400px' : '200px')};
  padding: 15px;
  transition: height .2s ease-in;
`

const Recipe = ({ handleClick, isSelected, recipe, index}) => (
  <StyledRecipe isSelected={isSelected} onClick={() => handleClick(index)}>
    <RecipeTitle text={recipe.title} />
    <ObliqueText text={`Serves ${recipe.servings}`} />
    <ul>
      {recipe.ingredients.map((ingredient, ingredientIndex) => (
        <li key={ingredientIndex}>
          {ingredient.quantity} {ingredient.unit} {ingredient.ingredient}
        </li>
      ))}
    </ul>
    <br />
    <p>{recipe.description}</p>
  </StyledRecipe>
)

Recipe.defaultProps = {
  recipe: {
    title: '',
    servings: 0,
    ingredients: [],
    description: ''
  }
}

export default Recipe
