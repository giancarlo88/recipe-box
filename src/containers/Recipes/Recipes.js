import React, { Component } from 'react'
import styled from 'styled-components'

import Recipe from '../../components/Recipe/Recipe'

export default class Recipes extends Component {
  render() {
    return <Recipe/>
  }
}
