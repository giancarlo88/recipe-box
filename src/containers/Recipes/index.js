import React, { Component } from 'react'

import Recipe from '../../components/Recipe'

export default class Recipes extends Component {
  constructor() {
    super()
    this.state = {
      recipes: [],
      selectedRecipe: null
    }
  }

  handleRecipeClick = (index) => {
    this.setState((prevState) => {
      if (prevState.selectedRecipe !== index) {
        return { selectedRecipe: index }
      }
      return { selectedRecipe: null }
    })
  }

  async componentDidMount() {
    try {
      const res = await fetch('http://www.ggalliani.com/test/recipes.json')
      const recipes = await res.json()
      this.setState({
        recipes
      })
    } catch (error) {
      throw error
    }
  }

  render() {
    return this.state.recipes.map((recipe, index) => (
      <Recipe
        key={index}
        index={index}
        recipe={recipe}
        isSelected={index === this.state.selectedRecipe}
        handleClick={(index) => this.handleRecipeClick(index)}
      />
    ))
  }
}
