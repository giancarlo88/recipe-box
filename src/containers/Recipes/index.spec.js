import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import Recipes from '.'
import Recipe from '../../components/Recipe'

describe('<Recipes />', () => {
  const recipes = [
    {
      title: 'Test recipe 1',
      servings: 4,
      ingredients: [
        {
          quantity: 1,
          unit: 'cup',
          ingredient: 'flour'
        }
      ],
      description: 'A test recipe',
      notes: 'It tastes like Mocha.'
    }
  ]

  fetch.mockResponse(JSON.stringify(recipes))

  beforeEach(() => {
    jest.resetModules()
  })

  const mockRender = async () => {
    const component = mount(<Recipes />)
    await component.instance().componentDidMount()
    component.update()
    return component
  }

  it('should match a snapshot', async () => {
    const tree = create(<Recipes />)
    await tree.getInstance().componentDidMount()
    expect(tree).toMatchSnapshot()
  })

  it('should pass props to the Recipe component', async () => {
    const component = await mockRender()
    const child = component.find(Recipe)
    expect(child.props().index).toBe(0)
    expect(child.props().recipe).toEqual(recipes[0])
    expect(child.props().isSelected).toBe(false)
  })

  it('should update the isSelected index when one of the recipes are clicked', async () => {
    const component = await mockRender()
    expect(component.state().selectedRecipe).toBe(null)
    const child = component.find(Recipe)
    child.simulate('click')
    expect(component.state().selectedRecipe).toBe(0)
    expect(component.find(Recipe).props().isSelected).toBe(true)
  })
})
