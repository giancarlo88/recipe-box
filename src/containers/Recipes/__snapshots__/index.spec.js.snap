// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipes /> should match a snapshot 1`] = `
<div
  className="sc-htpNat jdaKho"
  onClick={[Function]}
>
  <h2
    className="sc-bwzfXH cIKGFn"
  >
    Test recipe 1
  </h2>
  <p
    className="sc-bdVaJa hCJUJ"
  >
    Serves 4
  </p>
  <ul>
    <li>
      1
       
      cup
       
      flour
    </li>
  </ul>
  <br />
  <p>
    A test recipe
  </p>
</div>
`;
